### PR TITLE
app.nit: navigate between windows with the back button

### PR DESCRIPTION
**Pull request by xymus**
This should be the last big feature of _app.nit_ needed for my thesis!

Intro services to navigate between multiple windows of the same app.
Windows are added to a stack and popped back to the visible state on pressing the back key.
This behavior was implemented for Android by intercepting events raised by the "hardware" back key.
iOS offers the same features natively so there is no adaptation needed as of now.
On GNU/Linux with GTK+, a button is added to the window header when there's a window to go back to.

These feature will probably have to be tweaked in the future, but they are enough for the Benilux client app. Notably, the life-cylce of each individual window has to be reconsidered and we will probably need some adaptation on iOS to have a better integration.

This PR also fixes a few bugs. The `EditText` losing focus in a `ListLayout` is fixed by a change to the Android manifest file, and vertical layout should now look better on iOS.

I took the opportunity to add other key events while working on the Android back key support.
These will be useful when migrating games from the old low-level implementation on the NDK to our custom `NitActivity.java`.

### CODE REVIEW

**Comment by privat** [comment]
+1

**Comment by xymus** [comment]
test osx please

File: lib/android/nit_activity.nit [hunk 1]
```diff
@@ -279,6 +286,11 @@ class Activity
 	# Notification from Android, the current device configuration has changed
 	fun on_configuration_changed do end
 
+	# The back key has been pressed
+	#
+	# Return `true` if the event has been handled.
+	fun on_back_pressed: Bool do return false
```
Comment (BlackMinou) [inline lib/android/nit_activity.nit]: why does this return false by default ? if it's meant to be refined, why not make this function abstract ?

**Comment by BlackMinou** [comment]
late +1, but +1 anyway

File: lib/android/nit_activity.nit (hunk 1 above)
Comment (xymus) [inline lib/android/nit_activity.nit]: The doc could be clearer... It's the "stardard" way of doing this in the app.nit framework.  The return value is passed to the underlying Java code. Refinements can return true to intercept the event, but the default returns false and let the OS do its thing. In this case it's handle the back button and close the app.

Comment (BlackMinou) [inline lib/android/nit_activity.nit]: Oh yeah I forgot about it! nevermind then

**Pull request merged**